zip.ml

```diff
@@ -588,8 +588,16 @@ let add_entry_header ofile comment level mtime filename =
     raise(Error(ofile.of_filename, filename, "wrong compression level"));
   if String.length filename >= 0x10000 then
     raise(Error(ofile.of_filename, filename, "filename too long"));
+  if not (Filename.is_relative filename) then
+    raise(Error(ofile.of_filename, filename, "file name must not be absolute"));
   if String.length comment >= 0x10000 then
     raise(Error(ofile.of_filename, filename, "comment too long"));
+  let filename =
+    if not (String.contains filename '\\') then filename
```

Comment:
This implements the constraint from the spec, but I wonder if it could break some existing uses of the library. I tried under Linux, and zip/unzip don't complain about backslashes in file names (i.e. they don't respect the spec). One possibility could be to keep backslashes in filenames when not running on Windows; it's not nice, and the spec about the file format is not supposed to depend on which OS we used to create the file.

If we keep the strict semantics, I think it should be documented in the module interface and Changelog (as a possibly breaking changes for non-Windows users).

Another point: what about `Sys.os_type = "Cygwin"`?  In that case, one might also expect backslashes in filenames manipulated by the application.

(FTR, on Windows, unzipping a file with entry containing backslashes behaves as follows:
  - Using Cygwin's `unzip`: backslashes are turned into underscores.
  - Using WSL `unzip` (on an NTFS drive): backslashes are turned into the Unicode Private Use Area U+F05C.
)